### PR TITLE
Adding support for Zstandard compression

### PR DESCRIPTION
**Pull request by clavin-xlnx**
[Zstandard](http://facebook.github.io/zstd/) is a compression format that has some additional benefits that we plan to adopt and deploy in RapidWright.  RapidWright has a number of large data files that are already compressed using `gzip` compression.  However, Zstandard has shown that it can improve the compression ratio and also load files faster than the defacto method.  

More details on the impact of Zstandard adoption will be included in with the next release.

### CODE REVIEW

File: LICENSE.TXT [hunk 1]
```diff
@@ -102,7 +102,8 @@ RapidWright (Apache 2.0)
   - JetBrains Java Annotations 20.1.0 (Apache 2.0)
   - jzlib 1.1.3 (BSD-style License)
   - Apache Commons IO 2.11.0 (Apache 2.0)
-
+  - Zstd-jni 1.5.5-1 (2-clause BSD)
+      - ZStandard 1.5.5 (3-clause BSD or GPLv2)
```
Comment (eddieh-xlnx) [inline LICENSE.TXT:106]:
```suggestion
  - Zstd-jni 1.5.5-1 (2-clause BSD)
  - Zstandard 1.5.5 (3-clause BSD or GPLv2)

```

File: src/com/xilinx/rapidwright/util/FileTools.java [hunk 2]
```diff
@@ -175,40 +177,174 @@ public class FileTools {
...
     public static Output getKryoOutputStreamWithoutDeflater(OutputStream os) {
         return useUnsafeStreams() ? new UnsafeOutput(os)
                                   : new Output(os);
     }
 
+    /**
+     * Creates a Kryo input stream that instantiates a Zstandard compression stream
+     * to an input file.
+     * 
+     * @param fileName Name of the file to target.
```
Comment (eddieh-xlnx) [inline src/com/xilinx/rapidwright/util/FileTools.java:278]:
```suggestion
     * Creates a Kryo input stream from decompressing a Zstandard compressed input
     * file
     * 
     * @param fileName Name of the file to read from.
```
I think many of the "input" variants below this one could also be reworded to be clearer?

**Review by eddieh-xlnx — APPROVED** [review]

File: LICENSE.TXT [hunk 3]
```diff
@@ -102,7 +102,8 @@ RapidWright (Apache 2.0)
   - JetBrains Java Annotations 20.1.0 (Apache 2.0)
   - jzlib 1.1.3 (BSD-style License)
   - Apache Commons IO 2.11.0 (Apache 2.0)
-
+  - Zstd-jni 1.5.5-1 (2-clause BSD)
+      - Zstandard 1.5.5 (3-clause BSD or GPLv2)
```
Comment (eddieh-xlnx) [inline LICENSE.TXT:106]: Spacing still wonky here?

Comment (clavin-xlnx) [reply LICENSE.TXT:106]: Not sure what you mean.  Zstd-jni depends on Zstandard, this is just an indent (see examples above).

Comment (eddieh-xlnx) [reply LICENSE.TXT:106]: Oooh, I see! I didn't notice the indent out of my view. Sorry!

**Review by eddieh-xlnx — APPROVED** [review]

**Pull request merged**